hazelcast/src/main/java/com/hazelcast/spi/impl/sequence/CallIdSequenceWithBackpressure.java

```diff
@@ -94,13 +94,20 @@ private boolean hasSpace() {
         return longs.get(INDEX_HEAD) - longs.get(INDEX_TAIL) < maxConcurrentInvocations;
     }
 
-    private void waitForSpace() throws TimeoutException {
-        long deadline = System.nanoTime() + backoffTimeoutNanos;
+    private void waitForSpace() {
+        if (backoffTimeoutNanos <= 0) {
+            throw new HazelcastOverloadException(String.format(
+                    "Maximum invocation count reached." + " maxConcurrentInvocations = %d, backoffTimeout = %d msecs",
+                    maxConcurrentInvocations, NANOSECONDS.toMillis(backoffTimeoutNanos)));
+        }
+
+        long start = System.nanoTime();
         for (long idleCount = 0; ; idleCount++) {
-            if (System.nanoTime() >= deadline) {
-                throw new TimeoutException(String.format("Timed out trying to acquire another call ID."
-                        + " maxConcurrentInvocations = %d, backoffTimeout = %d", maxConcurrentInvocations,
-                        NANOSECONDS.toMillis(backoffTimeoutNanos)));
+            long elapsedNanos = System.nanoTime() - start;
+            if (elapsedNanos > backoffTimeoutNanos) {
+                throw new HazelcastOverloadException(String.format("Timed out trying to acquire another call ID."
+                        + " maxConcurrentInvocations = %d, backoffTimeout = %d msecs, elapsed:%d msecs", maxConcurrentInvocations,
```

Comment:
What is the added value of adding elapsed? elapsed = backoff timeout.

Reply:
It is just an additional information for debugging. E.g. If the elapsed is too higher than the backofftimeout, may be there is a problem with backoff. It is usually helpful to put the actual values in debugging log.